Cleanup and add more samples for ECS/EC2 log collection (#16087)
DataDog/documentation: content/en/containers/amazon_ecs/logs.md

You can consult [the Docker Log Collection setup instructions][12] for information on how to use Autodiscovery configurations. For example a log configuration like the following can be used to override the `source` and `service` of the logs collected:

```json
[{"source": "example-source", "service": "example-service"}]
```

With respect to ECS, this can be added to the label `com.datadoghq.ad.logs` within the `dockerLabels` of the Task Definition for the application container emitting those logs.

```json
{
  "containerDefinitions": [
    {
      "name": "<CONTAINER_NAME>",
      "image": "<CONTAINER_IMAGE>",
      "dockerLabels": {
        "com.datadoghq.ad.logs": "[{\"source\": \"example-source\", \"service\": \"example-service\"}]"
      }
    }
  ]
}
```

You can further customize this by [adding tags to your log configuration][4] or any `log_processing_rules` for [advanced log collection options][5].

### Log file within a container

> Docker with the `default` or `json-file` driver exposes the `stdout` and `stderr` log streams in a format that the Agent can readily find. However, if a container is creating a log file isolated within its container then the Agent does not natively have visibility to that file. Datadog recommends using the `stdout` and `stderr` output streams for containerized applications when possible, to more automatically set up log collection. If not possible, this can be done by providing an Autodiscovery log configuration to point towards the desired file path. While also having the Agent container and application container share the directory on the host containing the log file.

Reviewer: ```suggestion
Docker (with the `default` or `json-file` driver) exposes the `stdout` and `stderr` log streams in a format that the Agent can readily find. However, if a container is creating a log file isolated within its container, then the Agent does not natively have visibility to that file. Datadog recommends using the `stdout` and `stderr` output streams for containerized applications to more automatically set up log collection. If this is not possible, you can provide an Autodiscovery log configuration to point towards the desired file path, and ensure that the Agent container and application container share a directory on the host containing the log file.
```

> The log configuration below can be provided to tell the Agent to [collect this custom log file][3] at the `/var/log/example/app.log` path.

Reviewer: ```suggestion
The log configuration below tells the Agent to [collect this custom log file][3] at the `/var/log/example/app.log` path.
```

```json
[{
  "type": "file",
  "path": "/var/log/example/app.log",
  "source": "example-source",
  "service": "example-service"
}]
```

> As an example the Task Definition below is

Reviewer: ```suggestion
Example: the Task Definition below performs the following:
```

> 1. Writing some logs to that file `/var/log/example/app.log` file
> 2. Has the `dockerLabels` present to setup the log configuration
> 3. Has the host path `volumes` and `mountPoints` specified for this `/var/log/example` directory

Reviewer: ```suggestion
* Writing some logs to that file `/var/log/example/app.log` file
* Has the `dockerLabels` present to setup the log configuration
* Has the host path `volumes` and `mountPoints` specified for this `/var/log/example` directory
```

```json
{
  "containerDefinitions": [
    {
      "name": "example-logger",
      "image": "busybox",
      "entryPoint": ["/bin/sh", "-c", "--"],
      "command": ["while true; do sleep 1; echo `date` example file log >> /var/log/example/app.log; done;"],
      "mountPoints": [
        {
          "containerPath": "/var/log/example",
          "sourceVolume": "applogs"
        }
      ],
      "dockerLabels": {
        "com.datadoghq.ad.logs": "[{\"type\":\"file\",\"path\":\"/var/log/example/app.log\",\"source\":\"example-source\",\"service\":\"example-service\"}]"
      }
    }
  ],
  "volumes": [
    {
      "host": {
        "sourcePath": "/var/log/example"
      },
      "name": "applogs"
    }
  ],
  "family": "example-logger"
}
```

> The file paths for the configuration are always relative to the Agent. So that same `volume` and `mountPoint` need to be present within the Agent's Task Definition as well to give visibility to that log file.

Reviewer: ```suggestion
The file paths for the configuration are always relative to the Agent. The same `volume` and `mountPoint` also need to be present within the Agent's Task Definition to give visibility to that log file.
```

See the [AWS Bind mounts documentation][6] for additional details on volume management with ECS.

> **Note**: When using this kind of configuration with a container the `stdout` and `stderr` logs streams are not collected automatically from the container, only the file. If collection from both the container streams and a file are needed it should be explicitly enabled in the configuration. For example:

Reviewer: ```suggestion
**Note**: When using this kind of configuration with a container, the `stdout` and `stderr` logs streams are not collected automatically from the container—only the file. If collection from both the container streams and a file are needed, explicitly enable this in the configuration. For example:
```
